src/PostgREST/SchemaCache.hs

```diff
@@ -697,17 +705,19 @@ tablesSqlQuery pgVer =
   "ORDER BY table_schema, table_name"
   where
     relIsPartition = if pgVer >= pgVersion100 then " AND not c.relispartition " else mempty
-    -- detect default values on columns that have GENERATED .. AS IDENTITY
-    columnDefault =
-      if pgVer >= pgVersion100
-        then [q|
-        CASE
-           WHEN nullif(a.attidentity, '') is null
-             THEN pg_get_expr(ad.adbin, ad.adrelid)::text
-             ELSE format('nextval(%s)', quote_literal(pg_get_serial_sequence(a.attrelid::regclass::text, a.attname::text)))
-        END AS column_default,|]
-        else "pg_get_expr(ad.adbin, ad.adrelid)::text AS column_default,"
-
+    columnDefault
+      | pgVer < pgVersion100                          = "pg_get_expr(ad.adbin, ad.adrelid)::text as column_default,"
+      | pgVer >= pgVersion100 && pgVer < pgVersion120 = [q|
+          CASE
+            WHEN a.attidentity = 'd' THEN format('nextval(%s)', quote_literal(seqsch.nspname || '.' || seqclass.relname))
+            ELSE pg_get_expr(ad.adbin, ad.adrelid)::text
+          END AS column_default,|]
+      | otherwise = [q|
+          CASE
+            WHEN a.attidentity  = 'd' THEN format('nextval(%s)', quote_literal(seqsch.nspname || '.' || seqclass.relname))
+            WHEN a.attgenerated = 's' THEN null
+            ELSE pg_get_expr(ad.adbin, ad.adrelid)::text
+          END AS column_default,|]
```

Comment:
```suggestion
    columnDefault
      | pgVer >= pgVersion120 = [q|
          CASE
            WHEN a.attidentity  = 'd' THEN format('nextval(%s)', quote_literal(seqsch.nspname || '.' || seqclass.relname))
            WHEN a.attgenerated = 's' THEN null
            ELSE pg_get_expr(ad.adbin, ad.adrelid)::text
          END AS column_default,|]
      | pgVer >= pgVersion100 = [q|
          CASE
            WHEN a.attidentity = 'd' THEN format('nextval(%s)', quote_literal(seqsch.nspname || '.' || seqclass.relname))
            ELSE pg_get_expr(ad.adbin, ad.adrelid)::text
          END AS column_default,|]
      | otherwise  = "pg_get_expr(ad.adbin, ad.adrelid)::text as column_default,"
```